lib/Dialect/TTNN/Analysis/ShardSolver.cpp

```diff
@@ -508,13 +508,6 @@ bool ShardSolver::checkShardCompatible(
   // TEMP : Dummy mock implementation, will be replaced.
   //
 
-  if (OpModel backend = dyn_cast<OpModel>(consumerOp)) {
```

Comment:
How should call roughly look?

Reply:
Its ranked tensor type now or shape plus layout?

Reply:
it's ranked tensor type now. underlying API to TTNNModelLib is shape plus layout. 

Do you have preference? 

People on the meeting yesterday didn't have a strong opinion.